content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,256 @@
+
+Once you're done with configuration, publish your Worker using Wrangler. 
+
+h2. Configuring Ably Reactor
+
+After publishing your Worker to your Cloudflare account, it will have a URL, you will need this to set up the Ably Reactor.
+
+Go to the dashboard for the Ably app you're using, choose the "Reactor" tab, then click "New Reactor Rule":
+
+<a href="/images/tutorials/reactor/reactor-event-rule-1.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-rule-1.png" style="width: 100%" alt="Create a new Reactor rule">
+</a>
+
+Choose the "Reactor Event" option:
+
+<a href="/images/tutorials/reactor/reactor-event-rule-2.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-rule-2.png" style="width: 100%" alt="Select the Reactor Event option">
+</a>
+
+Choose the "Cloudflare Workers" option:
+
+<a href="/images/tutorials/reactor/reactor-event-rule-cloudflare.png" target="_blank">  
+  <img src="/images/tutorials/reactor/reactor-event-rule-cloudflare.png" style="width: 100%" alt="Select the Cloudflare Workers option">
+</a>
+
+Fill out the form to provide Ably with the details it needs to invoke your Cloudflare Worker.
+
+Your Reactor event will now be created, and you'll be able to continue developing the game.
+
+h2. Publishing to an Ably Channel when a delivery is made
+
+Open the file *./src/feature/player/movement.js*
+
+This file contains the React code for handling player movements, but it also contains a connection to our Ably Channel, firstly, on line 18
+
+```[js]
+    window.Ably.connection.on(function(stateChange) {
+        console.log('New connection state is ' + stateChange.current)
+    })
+
+    const outboundChannel = window.Ably.channels.get('cloudflareworkerdemo')
+```
+
+This code establishes a connection to our "cloudflareworkerdemo" channel.
+
+When the player completes a level the `handleWin` function is called, and on line 64 we publish a message to our channel.
+
+```[js]
+  outboundChannel.publish('player', { playerSteps },
+      err => {
+          if (err) {
+              return console.error('Failed to publish', err)
+          }
+          console.log('published')
+      }
+  )
+```
+
+Our message contains the number of moves the player used to deliver the parcel.
+
+h2. Processing the message in the Worker
+
+Ably Reactor is monitoring the channel and calls the Cloudflare Worker with the message contents.
+We can see how this works by taking a look at the Woker file again *./cloudflare-worker/index.js*
+
+```[js]
+async function publishToAbly(data) {
+    try {
+        const URL = `https://rest.ably.io/channels/cloudflare-bot/messages?key=${key}`
+        
+        await fetch(`${URL}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ data })
+        });
+
+    } catch (error) {
+        return error
```

Comment:
Inconsistent use of semicolons.  On some lines it's used, on others not.  Please be consistent.